api/feedbooks.py

```diff
@@ -0,0 +1,457 @@
+            return RightsStatus.UNKNOWN
+
+        if rights in cls.RIGHTS_DICT:
+            # The CC license of the underlying text means it cannot be
+            # relicensed CC-BY-NC.
+            return cls.RIGHTS_DICT[rights]
+
+        # The default license as per our agreement with FeedBooks.
+        return RightsStatus.CC_BY_NC
+
+    @classmethod
+    def can_rehost_us(cls, rights, source, publication_year):
+        """Can we rehost this book on a US server?
+
+        :param rights: What FeedBooks says about the public domain status
+        of the book.
+
+        :param source: Where FeedBooks got the book.
+
+        :param publication_year: When the text was originally published.
+
+        :return: True if we can rehost in the US, False if we can't,
+        None if we're not sure. The distinction between False and None
+        is only useful when making lists of books that need to have
+        their rights status manually investigated.
+        """
+        if publication_year and publication_year < cls.PUBLIC_DOMAIN_CUTOFF:
+            # We will rehost anything published prior to 1923, no
+            # matter where it came from.
+            return True
+
+        if rights in cls.CAN_REHOST_IN_US:
+            # This book's FeedBooks rights statement explicitly marks
+            # it as one that can be rehosted in the US.
+            return True
+
+        # The rights statement isn't especially helpful, but maybe we
+        # can make a determination based on where Feedbooks got the
+        # book from.
+        source = (source or "").lower()
+
+        if any(site in source for site in cls.US_SITES):
+            # This book originally came from a US-hosted site that
+            # specializes in open-access books, so we must be able
+            # to rehost it.
+            return True
+
+        if source in ('wikisource', 'gutenberg'):
+            # Presumably en.wikisource and Project Gutenberg US.  We
+            # special case these to avoid confusing the US versions of
+            # these sites with other countries'.
+            return True
+
+        # And we special-case this one to avoid confusing Australian
+        # Project Gutenberg with US Project Gutenberg.
+        if ('gutenberg.net' in source and not 'gutenberg.net.au' in source):
+            return True
+
+        # Unless one of the above conditions is met, we must assume
+        # the book cannot be rehosted in the US.
+        if rights == cls.RIGHTS_UNKNOWN:
+            # To be on the safe side we're not going to host this
+            # book, but we actually don't know that it's unhostable.
+            return None
+
+        # In this case we're pretty sure. The rights status indicates
+        # some kind of general incompatible restriction (such as
+        # Life+70) and it's not a pre-1923 book.
+        return False
+
+class FeedbooksImportMonitor(OPDSImportMonitor):
```

Comment:
This class is new -- it replaces the behavior of a custom script in content with a customization of `OPDSImportMonitor`, which can be invoked with `OPDSImportScript`.